### ISSUE TITLE
Collect crashes when Try a demo project after deleting all projects on Android 5.1

### ISSUE THREAD

**Issue by kkrawczyk123** [getodk/collect#4870]
#### Software and hardware versions 
Collect v current master Android v 5.1, device used...

#### Problem description
When testing #4855 I have noticed crash on Android 5.1 device only when deleted all projects and then run demo project.

#### Steps to reproduce the problem
1. Create project
2. Delete project
3. Click on Try a demo

#### Expected behavior
Collect should not crash.

#### Other information 
stacktrace:
`2021-10-19 10:34:40.451 10957-10957/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.odk.collect.android, PID: 10957
    android.database.sqlite.SQLiteDiskIOException: disk I/O error (code 1802)
        at android.database.sqlite.SQLiteConnection.nativeExecuteForCursorWindow(Native Method)
        at android.database.sqlite.SQLiteConnection.executeForCursorWindow(SQLiteConnection.java:845)
        at android.database.sqlite.SQLiteSession.executeForCursorWindow(SQLiteSession.java:836)
        at android.database.sqlite.SQLiteQuery.fillWindow(SQLiteQuery.java:62)
        at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:144)
        at android.database.sqlite.SQLiteCursor.getCount(SQLiteCursor.java:133)
        at org.odk.collect.android.database.instances.DatabaseInstancesRepository.getCountByStatus(DatabaseInstancesRepository.java:109)
        at org.odk.collect.android.formmanagement.InstancesAppState.update(InstancesAppState.kt:36)
        at org.odk.collect.android.activities.viewmodels.MainMenuViewModel.lambda$refreshInstances$0$MainMenuViewModel(MainMenuViewModel.java:121)
        at org.odk.collect.android.activities.viewmodels.MainMenuViewModel$$ExternalSyntheticLambda1.get(Unknown Source)
        at org.odk.collect.async.CoroutineScheduler$immediate$1$result$1.invokeSuspend(CoroutineScheduler.kt:16)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)`

**Comment by seadowg** [comment]
This is a really weird one. It looks like the steps produce a crash on v2021.2.4 as well. In current `master` the app crashes but then recovers on the next run while in v2021.2.4 the user is forced wipe data/reinstall to get things working again so it looks like we've at least improved the situation (most likely with the change to creating dirs on demand). 

I've had a quick dig into this and the root problem seems to be that directories can't be created after they've been deleted - this wouldn't effect deleting a project and then recreating as it would have a different path. I've no idea why that's the case on 5.1 though. It doesn't appear to be a timing issue but it does appear that directories can be created successfully after the process is restarted (like happens via the crash). I was able to confirm this by hacking the code so we got a crash after project deletion.

It looks like we are seeing this happen in production ([crash logs from the v2021.2.4](https://console.firebase.google.com/u/0/project/api-project-322300403941/crashlytics/app/android:org.odk.collect.android/issues/14d430f22b6213f73b8e2f6b03f1a23b?time=last-ninety-days&versions=v2021.2.4%20(4267)&sessionEventKey=61686AE6011A00017C9382B3B1C6A706_1597424541112653051&os=5.1.1,5.0.2,5.0.1,5.0,5.1)) but as it's limited to just the demo project and a small section of users (Android 5/5.1 looks like it makes up just under 5%) I think we can maybe mark this as won't fix seeing as v2021.3 will make this recoverable. Another option would be to add a fail-safe that shows an error and asks the user to restart the app.

@kkrawczyk123 are we able to validate (as part of regression testing) that this definitely doesn't occur on any other OS levels?
